airflow/stats.py

```diff
@@ -159,35 +163,37 @@ class Timer(TimerProtocol):
     _start_time: int | None
     duration: int | None
 
-    def __init__(self, real_timer=None):
+    def __init__(self, real_timer: Timer | None = None) -> None:
         self.real_timer = real_timer
 
-    def __enter__(self):
+    def __enter__(self) -> Timer:
         return self.start()
 
-    def __exit__(self, exc_type, exc_value, traceback):
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
         self.stop()
 
-    def start(self):
+    def start(self) -> Timer:
         """Start the timer."""
         if self.real_timer:
             self.real_timer.start()
-        self._start_time = time.perf_counter()
+        self._start_time = int(time.perf_counter())
         return self
 
-    def stop(self, send=True):
+    def stop(self, send: bool = True) -> None:
         """Stop the timer, and optionally send it to stats backend."""
-        self.duration = time.perf_counter() - self._start_time
+        self.duration = int(time.perf_counter()) - (self._start_time or 0)
```

Comment:
This limits all timers to second gruanularity in output -- I don't think we wanted to do this do we?

Reply:
I assume it wasn't intentional, and was driven by the fact `_start_time` was originally typed as int. #30532 changes it back.

Reply:
Yeah, what Jed said.  That was to make MyPy happy due to the underlying typing for `_start_time`.  Thanks for fixing that, I should have questioned it instead of just rolling with it.